File: cpp/Array/L_largestPositiveNoExstWithNeg.cpp
#include <bits/stdc++.h>
using namespace std;

int findMaxK(std::vector<int> &nums)
{
    std::sort(nums.begin(), nums.end());
    int left = 0, right = nums.size() - 1;
    int maxK = std::numeric_limits<int>::min();
    while (left < right)
    {
        int sum = nums[left] + nums[right];
        if (sum == 0)
        {
            maxK = std::max(maxK, nums[right]);
            left++;
            right--;
        }
        else if (sum < 0)
        {
            left++;
        }
        else
        {
            right--;
        }
    }
    return maxK != std::numeric_limits<int>::min() ? maxK : -1; // If no such pair found
}

File: Python/Arrays/L_largestPositiveNoExstWithNeg.py
from typing import List


def findMaxK(nums: List[int]) -> int:
    n_set = set(nums)
    max_num = -1
    for n in n_set:
        if n * -1 in n_set:
            max_num = max(max_num, abs(n))
    return max_num